File: cpp/44d_1650A_DeletionsOfTwoAdjacentLetters.cpp
#include<bits/stdc++.h>

int main(){
    int t, flag;
    std::vector<std::string> ans;
    std::string s;
    char c;
    std::cin>>t;
    for(int i = 0; i < t; i++){
        std::cin>>s;
        std::cin>>c;
        flag = 0;
        if(s.size() == 1){
            if(s[0] == c){
                flag = 1;
            }
            else{
                flag = 0;
            }
        }
        else if(s.size() == 2){
            flag = 0;
        }
        else{
            for(int j = 0; j < s.size(); j++){
                if (s[j] == c && j  % 2 == 0){
                    flag = 1;
                    break;
                }
            }
        }
        if(flag == 0){
            ans.push_back("NO");
        }
        else{
            ans.push_back("YES");
        }
    }
    for (auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}

File: python/44d_1650A_DeletionsOfTwoAdjacentLetters.py
def main():
    ans = []
    t = int(input())
    for _ in range(t):
        s = input().strip()
        c = input()
        if len(s) == 2:
            ans.append("NO")
        else:
            flag = 0
            for i in range(len(s)):
                if s[i] == c and i % 2 == 0:
                    flag = 1
                    break
            if flag == 1:
                ans.append("YES")
            else:
                ans.append("NO")
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()